Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -471,22 +474,36 @@ inline void ClearPath() {
     }
 
     inline bool ShouldDraw() {
-        return CurrentGState().ShouldDraw();
+        return SUCCEEDED(_firstErrorHr) && CurrentGState().ShouldDraw();
     }
 
     inline void PushBeginDraw() {
         if ((_beginEndDrawDepth)++ == 0) {
-            deviceContext->BeginDraw();
+            if (__builtin_expect(SUCCEEDED(_firstErrorHr), true)) {
```

Comment:
other than the ICE, there is really no need to use __builtin_expect to optimize here - this is not a hot code path :).